api/controller/identity_url.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2012-2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package controller
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/params"
+)
+
+// IdentityProviderURL returns the URL of the configured external identity
+// provider for this controller or an empty string if no external identity
+// provider has been configured when the controller was bootstrapped.
```

Comment:
Can you set the external identity url using controller-config?

Reply:
AFAICT, you can only specify this when bootstrapping and then becomes read-only (see [here](https://github.com/juju/juju/blob/2fb13bbdc738665f08e6f98e7da5f692c1340cbc/controller/config.go#L279))

Reply:
TIL'd